pyth-sdk-solana/src/state.rs

```diff
@@ -280,49 +280,55 @@ pub struct Rational {
 #[repr(C)]
 pub struct PriceAccount {
     /// pyth magic number
-    pub magic:      u32,
+    pub magic:          u32,
     /// program version
-    pub ver:        u32,
+    pub ver:            u32,
     /// account type
-    pub atype:      u32,
+    pub atype:          u32,
     /// price account size
-    pub size:       u32,
+    pub size:           u32,
     /// price or calculation type
-    pub ptype:      PriceType,
+    pub ptype:          PriceType,
     /// price exponent
-    pub expo:       i32,
+    pub expo:           i32,
     /// number of component prices
-    pub num:        u32,
+    pub num:            u32,
     /// number of quoters that make up aggregate
-    pub num_qt:     u32,
+    pub num_qt:         u32,
     /// slot of last valid (not unknown) aggregate price
-    pub last_slot:  u64,
+    pub last_slot:      u64,
     /// valid slot-time of agg. price
-    pub valid_slot: u64,
+    pub valid_slot:     u64,
     /// exponentially moving average price
-    pub ema_price:  Rational,
+    pub ema_price:      Rational,
     /// exponentially moving average confidence interval
-    pub ema_conf:   Rational,
+    pub ema_conf:       Rational,
+    /// unix timestamp of aggregate price
+    pub timestamp:      i64,
+    /// min publishers for valid price
+    pub min_pub:        u8,
     /// space for future derived values
-    pub drv1:       i64,
+    pub drv2:           u8,
     /// space for future derived values
-    pub drv2:       i64,
+    pub drv3:           u16,
+    /// space for future derived values
+    pub drv4:           u32,
     /// product account key
-    pub prod:       AccKey,
+    pub prod:           AccKey,
     /// next Price account in linked list
-    pub next:       AccKey,
+    pub next:           AccKey,
     /// valid slot of previous update
-    pub prev_slot:  u64,
-    /// aggregate price of previous update
-    pub prev_price: i64,
-    /// confidence interval of previous update
-    pub prev_conf:  u64,
-    /// space for future derived values
-    pub drv3:       i64,
+    pub prev_slot:      u64,
+    /// aggregate price of previous update with TRADING status
+    pub prev_price:     i64,
+    /// confidence interval of previous update with TRADING status
+    pub prev_conf:      u64,
+    /// unix timestamp of previous aggregate with TRADING status
+    pub prev_timestamp: i64,
```

Comment:
Why is timestamp signed?

Reply:
It's mostly because Solana is using i64 to represent it and it is the same for unix system call of time (in new machines). 

It has benefit of easier time difference and also rust is careful about conversions.

In target chains (such as evm) it is implemented as the standard there.

Reply:
I added some comments to explain more.

Reply:
This stems from Solana using signed ints to represent timestamps :( 

- On Solana we will use signed, because they do.
- In the rust/terra/solana SDK we will use int, because they all use the same codebase. This isn't that big of a deal, because Rust's strongly typed system will prevent any mistakes.
- In the price attestation code we will use signed, to make the attester simple.
- In the EVM target chain contracts we will convert the timestamps to unsigned, because that's Solidity's native representation.

Alternative opinions/suggestions welcome 😄

Reply:
Thanks for helping me understand. 64-bit negative timestamps actually make sense, lots of stuff happened before 1970-01-01 :joy: However, shouldn't we go for the *_time thing here?